src/SprykerEco/Zed/Unzer/Business/Quote/UnzerKeypairQuoteExpander.php

```diff
@@ -104,7 +116,17 @@ protected function setRegularUnzerKeypair(QuoteTransfer $quoteTransfer): QuoteTr
     {
         $unzerCredentialsTransfer = $this->getRegularUnzerCredentials($quoteTransfer);
 
-        $quoteTransfer->getPaymentOrFail()->getUnzerPaymentOrFail()->setUnzerKeypair($unzerCredentialsTransfer->getUnzerKeypairOrFail());
+        if ($unzerCredentialsTransfer->getType() !== UnzerConstants::UNZER_CREDENTIALS_TYPE_STANDARD) {
+            $unzerMarketplacePaymentCredentialsFinderCriteriaTransfer = (new UnzerMarketplacePaymentCredentialsFinderCriteriaTransfer())
+                ->setQuote($quoteTransfer)
```

Comment:
You use the same object `QuoteTransfer` to get both criteria. One more point to not introduce `CriteriaTrasnfer`

Reply:
It's needed because one time its set by value from Quote and other time manually fro CreditCard